### PR TITLE
"Completed Unit Testing"

### PR DESCRIPTION
**Pull request by skorpion-g**

### CODE REVIEW

**Review by TolyK — COMMENTED** [review]
Looks good, nice update! For the catch error block, I'd add in the actual error somewhere either console.error(error) or ie for saveSong() return('Error in saving song to DB', error). I think this would help get more info for debugging if there was an error instead of a static console.log.

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "",
+  "name": "frankie",
```
Comment (LesliePajuelo) [inline package.json:2]: Not frankie-player?

File: seedingScript.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 const songData = require('./songData.js');
 
-var seederboi = () => {
+var seederboi = async () => {
```
Comment (LesliePajuelo) [inline seedingScript.js:3]: Why const and var? use let or only use var. var and (const and let) have different rules for scope.  mixed syntax invites bugs

File: server/index.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+app.use(express.static(path.join(__dirname + '../client')));
+app.use(bodyParser.urlencoded({ extended: false }))
+app.use(bodyParser.json())
+
+// INCLUDE "try" AND "catch" IN THE ASYNC AWAIT BELOW
+
+app.get('/songdata/:id', async (req, res) => {
+  try {
+    var id = req.params.id;
```
Comment (LesliePajuelo) [inline server/index.js:16]: What in the try will throw an error?

What is are you trying to do that might fail?

File: server/index.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    if (id > 100 || id < 0) {
+      res.end('SONG DOES NOT EXIST');
+    } else {
+      var findSongByID = await songData.findSong(id)
```
Comment (LesliePajuelo) [inline server/index.js:20]: What if there is an error from songData? this is what should be in the try block

File: server/index.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      res.send(findSongByID);
+    }
+  } catch (error) {
+    console.log(error)
```
Comment (LesliePajuelo) [inline server/index.js:24]: no error info for the client?

File: songData.js [hunk 6]
```diff
@@ -22,31 +22,47 @@ const songDataSchema = new mongoose.Schema({
...
-      callback(err);
-    } else {
-      console.log(`Song, "${song.songName}" saved to DB`);
-      callback(null, data);
-    }
-  })
+var saveSong = async (songData, callback) => {
+  try {
+    let song = new Song(songData);
+    var saved = await song.save(( ) => {
```
Comment (LesliePajuelo) [inline songData.js:28]: await in the try, good!

But wait? why are you returning to the saved variable? you're not sending it back. How would you send it back? would you want to? 

Are you going to use the callback?

**Comment by LesliePajuelo** [comment]
Remove your bundle.js from github.

File: test/serverSpec.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+})
+
+describe(`Shouldn't be able to save anything to DB from POST`, () => {
+  it('Should response with a 100 for bad request', (done) => {
+    request(app)
+    .post('/songdata')
+    .send({"id" : 1})
+    .expect(404)
+    .end((err, res) => {
+      if(err) {
```
Comment (LesliePajuelo) [inline test/serverSpec.js:76]: what? why have the if statement?

File: test/serverSpec.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+      .expect('Content-Type', "text/html; charset=utf-8")
+      .expect(200)
+    done();
+  })
+});
+
+describe(`Should only get data for valid id's`, () => {
+  it('Should respond with a 400 if incorrect song id', (done) => {
+    request(app)
+    .get('/songdata/' + 0)
```
Comment (LesliePajuelo) [inline test/serverSpec.js:57]: nitpicky, I know. but why not '/songdata/0'?

File: test/serverSpec.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    .post('/songdata')
+    .send({"id" : 1})
+    .expect(404)
+    .end((err, res) => {
+      if(err) {
+        done(err)
+      } else {
+        done(err)
+      }
+    })
```
Comment (LesliePajuelo) [inline test/serverSpec.js:81]: Just realized you're using red/green (TDD) for this. 🏆 ❗

**Review by LesliePajuelo — COMMENTED** [review]
Really, remove the bundle.js from gh

File: test/serverSpec.js (hunk 8 above)
Comment (skorpion-g) [reply test/serverSpec.js:57]: I don't have a good reason aside I thought it might throw me an error.

File: package.json (hunk 1 above)
Comment (skorpion-g) [reply package.json:2]: good catch! lemme fix that

**Pull request merged**